numba/targets/npyimpl.py

```diff
@@ -446,6 +446,25 @@ def lower_binary_operator(context, builder, sig, args):
     register(implement(operator, _arr_kind, _any)(lower_binary_operator))
 
 
+def register_inplace_binary_operator_kernel(operator, kernel):
+    def lower_binary_operator(context, builder, sig, args):
+        if len(args) == 2:
+            # Output is saved to the first operand
+            sig.args = list(sig.args)
+            sig.args.append(sig.args[0])
```

Comment:
Similarly, rather than mutating the signature, it sounds better to generate a new one. You can call `typing.signature()` for that.